scripts/envDFL.sh

```diff
@@ -494,120 +494,221 @@ build_tests_coverage() {
+  else
     if [ ! -f "$2" ]; then
-        error_exit "IIDM network file $1 doesn't exist"
+      error_exit "IIDM network file $1 doesn't exist"
     fi
     if [ ! -f "$3" ]; then
-        error_exit "DFL configuration file $2 doesn't exist"
+      error_exit "DFL configuration file $2 doesn't exist"
     fi
     gdb --args $DYNAFLOW_LAUNCHER_INSTALL_DIR/bin/DynaFlowLauncher \
     --log-level $DYNAFLOW_LAUNCHER_LOG_LEVEL \
     --network $2 \
     --config $3
+  fi
 }
 
 launch_valgrind() {
+  if [ -n "$ZIP_ARCHIVE" ]; then
+    valgrind --tool=callgrind --dump-instr=yes --collect-jumps=yes $DYNAFLOW_LAUNCHER_INSTALL_DIR/bin/DynaFlowLauncher \
+    --log-level $DYNAFLOW_LAUNCHER_LOG_LEVEL \
+    --network $2 \
+    --config $3 \
+    --input-archive $ZIP_ARCHIVE
+  else
     if [ ! -f "$2" ]; then
-        error_exit "IIDM network file $1 doesn't exist"
+      error_exit "IIDM network file $1 doesn't exist"
     fi
     if [ ! -f "$3" ]; then
-        error_exit "DFL configuration file $2 doesn't exist"
+      error_exit "DFL configuration file $2 doesn't exist"
     fi
     valgrind --tool=callgrind --dump-instr=yes --collect-jumps=yes $DYNAFLOW_LAUNCHER_INSTALL_DIR/bin/DynaFlowLauncher \
     --log-level $DYNAFLOW_LAUNCHER_LOG_LEVEL \
     --network $2 \
     --config $3
+  fi
 }
 
 
 launch_sa() {
-    if [ ! -f "$2" ]; then
+  export_preload
+  if [ "${DYNAWO_USE_MPI}" == "YES" ]; then
+    if [ -n "$ZIP_ARCHIVE" ]; then
+      "$MPIRUN_PATH" -np $NBPROCS $DYNAFLOW_LAUNCHER_INSTALL_DIR/bin/DynaFlowLauncher --log-level $DYNAFLOW_LAUNCHER_LOG_LEVEL \
+                                                                                      --network $2 \
+                                                                                      --config $3 \
+                                                                                      --contingencies $4 \
+                                                                                      --input-archive $ZIP_ARCHIVE
+    else
+      if [ ! -f "$2" ]; then
         error_exit "IIDM network file $2 doesn't exist"
-    fi
-    if [ ! -f "$3" ]; then
+      fi
+      if [ ! -f "$3" ]; then
         error_exit "DFL configuration file $3 doesn't exist"
-    fi
-    if [ ! -f "$4" ]; then
+      fi
+      if [ ! -f "$4" ]; then
         error_exit "Security Analysis contingencies file $4 doesn't exist"
-    fi
-    export_preload
-    if [ "${DYNAWO_USE_MPI}" == "YES" ]; then
+      fi
       "$MPIRUN_PATH" -np $NBPROCS $DYNAFLOW_LAUNCHER_INSTALL_DIR/bin/DynaFlowLauncher --log-level $DYNAFLOW_LAUNCHER_LOG_LEVEL \
                                                                                       --network $2 \
                                                                                       --config $3 \
                                                                                       --contingencies $4
+    fi
+  else
+    if [ -n "$ZIP_ARCHIVE" ]; then
```

Comment:
Maybe we should add a check that the zip archive exists

Reply:
The check is made in the C++ code

Reply:
yes, but it is also the case for network, config and contingencies files and yet they are tested in the other branch. This is to be homogeneous